### PR TITLE
Fix broadcast root during the replication call

### PR DESCRIPTION
**Pull request by jnke2016**
A Raft https://github.com/rapidsai/raft/pull/1573 assigning deterministic ranks to dask workers was merged, breaking batch algorithms like batch_edge_betweenness_centrality by picking the wrong worker as the root for the broadcast operation.
This PR ensures that the worker with rank = 0 is the root of the broadcast operation.

### CODE REVIEW

File: python/cugraph/cugraph/dask/common/input_utils.py
Comment (VibhuJawa) [inline python/cugraph/cugraph/dask/common/input_utils.py:109]: I think this will fail on MNMG setups. The `worker_rank`  here gets `nvml_device_index` which will be zero for 2 devices (the first device on both nodes) . 

I think we should figure out a way to get this from raft instead.

**Review by VibhuJawa — CHANGES_REQUESTED** [review]

File: python/cugraph/cugraph/dask/common/input_utils.py
Comment (VibhuJawa) [reply python/cugraph/cugraph/dask/common/input_utils.py:109]: How about we directly fetch from the upstream `raft_dask` library to prevent breaking in the future too ? 
```suggestion
            from raft_dask.common.comms import _func_worker_ranks
            worker_ranks = _func_worker_ranks(client)
```

**Review by VibhuJawa — CHANGES_REQUESTED** [review]

File: python/cugraph/cugraph/dask/common/input_utils.py
Comment (jnke2016) [reply python/cugraph/cugraph/dask/common/input_utils.py:109]: That PR is not merged yet. That is why I added a FIXME in line 286

Comment (jnke2016) [reply python/cugraph/cugraph/dask/common/input_utils.py:109]: > I think this will fail on MNMG setups. The worker_rank here gets nvml_device_index which will be zero for 2 devices (the first device on both nodes)

Why would it be zero? And I extensively tested this PR on 2 and 4 nodes and all runs passed

Comment (jnke2016) [reply python/cugraph/cugraph/dask/common/input_utils.py:109]: > That PR is not merged yet. That is why I added a FIXME in line 286
No need to wait for your PR to be merged because `_func_worker_ranks` already exits. Its implementation will just change once your PR is merge.

> I think this will fail on MNMG setups
I don't think this can fail because the address of rank 0 in this PR `always` always matches the raft one. In fact, `client.run(_get_nvml_device_index)` is deterministic and always returns the result in increasing network ID or IP address number(the keys of the dictionary). On a MNMG run, even though there are multiple `rank 0`, this PR will always pick the one in the first node which is consistent with the raft one. This is true because the raft PR only applies the rank offsets to the second node and above. Furthermore, I extensively tested this PR on Friday on both 2 and 4 nodes and all runs passed.


> How about we directly fetch from the upstream raft_dask library to prevent breaking in the future too ?
Right and this is what I had in mind when I added a FIXME. This will avoid code duplication and prevent breaking changes.

Comment (VibhuJawa) [reply python/cugraph/cugraph/dask/common/input_utils.py:109]: > On a MNMG run, even though there are multiple pairs with rank 0, this PR will always pick the one in the first node which is consistent with the raft one
Does this always hold true, Maybe i am missing the logic on this

File: python/cugraph/cugraph/dask/common/input_utils.py
Comment (jnke2016) [reply python/cugraph/cugraph/dask/common/input_utils.py:287]: Ya. I was pushing this now in another commit

**Review by VibhuJawa — APPROVED** [review]
LGTM

File: python/cugraph/cugraph/dask/common/input_utils.py
Comment (VibhuJawa) [reply python/cugraph/cugraph/dask/common/input_utils.py]: Approved

**Review by rlratzel — APPROVED** [review]

File: python/cugraph/cugraph/dask/common/input_utils.py
Comment (jnke2016) [reply python/cugraph/cugraph/dask/common/input_utils.py]: As we discussed, getting the ranks that were established after the comms initialization guarantee consistency.

**Pull request merged**
